app/api/items/start_result_path.go

```diff
@@ -0,0 +1,204 @@
+//   "403":
+//     "$ref": "#/responses/forbiddenResponse"
+//   "500":
+//     "$ref": "#/responses/internalErrorResponse"
+func (srv *Service) startResultPath(w http.ResponseWriter, r *http.Request) service.APIError {
+	var err error
+
+	ids, err := idsFromRequest(r)
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	participantID := service.ParticipantIDFromContext(r.Context())
+
+	var result []map[string]interface{}
+	apiError := service.NoError
+	var previousAttemptID int64
+	err = srv.Store.InTransaction(func(store *database.DataStore) error {
+		result = getDataForResultPathStart(store, participantID, ids)
+		if len(result) == 0 {
+			apiError = service.InsufficientAccessRightsError
+			return apiError.Error
+		}
+
+		data := result[0]
+		rowsToInsert := make([]map[string]interface{}, 0, len(data))
+		for index, itemID := range ids {
+			attemptID := data[fmt.Sprintf("attempt_id%d", index)]
+			if attemptID == nil {
+				attemptID = previousAttemptID
+			}
+			previousAttemptID = attemptID.(int64)
+			if data[fmt.Sprintf("has_started_result%d", index)].(int64) == 1 {
+				continue
+			}
+			rowsToInsert = append(rowsToInsert, map[string]interface{}{
+				"item_id":                  itemID,
+				"participant_id":           participantID,
+				"attempt_id":               attemptID,
+				"started_at":               database.Now(),
+				"latest_activity_at":       database.Now(),
+				"result_propagation_state": "to_be_propagated",
+			})
+		}
+		if len(rowsToInsert) > 0 {
+			resultStore := store.Results()
+			service.MustNotBeError(resultStore.InsertOrUpdateMaps(rowsToInsert,
+				[]string{"started_at", "latest_activity_at", "result_propagation_state"}))
+			service.MustNotBeError(resultStore.Propagate())
+		}
+
+		return nil
+	})
+	if apiError != service.NoError {
+		return apiError
+	}
+	service.MustNotBeError(err)
+
+	service.MustNotBeError(render.Render(w, r, service.UpdateSuccess(map[string]interface{}{
+		"attempt_id": strconv.FormatInt(previousAttemptID, 10),
+	})))
+	return service.NoError
+}
+
+func getDataForResultPathStart(store *database.DataStore, participantID int64, ids []int64) []map[string]interface{} {
+	participantAncestors := store.ActiveGroupAncestors().Where("child_group_id = ?", participantID).
+		Joins("JOIN `groups` ON groups.id = groups_ancestors_active.ancestor_group_id")
+	participantActivities := participantAncestors.Select("groups.root_activity_id").WithWriteLock()
+	participantSkills := participantAncestors.Select("groups.root_skill_id").WithWriteLock()
+
+	subQuery := store.Table("visible_items as items0").WithWriteLock().Where("items0.id = ?", ids[0]).
+		Where("items0.id IN ? OR items0.id IN ?",
+			participantActivities.SubQuery(), participantSkills.SubQuery())
+
+	previousAttemptIDForCondition := "attempts0.id"
+	var score string
+	var columns string
+	attemptIsActiveCondition := "1"
+	var columnsForOrder string
+	for i := 0; i < len(ids); i++ {
```

Comment:
So you add more join for each depth? Does it really scale? I am surprised you do not limit (to a quite "low" value) the max number of elements in the path. I cannot see how it cannot cause a problem is the user comes up with a path of 1000 elements. 
Would it be safer (even if probably slower in total on average) to do that iteratively?
(maybe I didn't understood correctly what you did)

Reply:
Actually there is a limit:
```
func idsFromRequest(r *http.Request) ([]int64, error) {
	ids, err := service.ResolveURLQueryPathInt64SliceField(r, "ids")
	if err != nil {
		return nil, err
	}
	if len(ids) > 10 {
		return nil, errors.New("no more than 10 ids expected")
	}
	return ids, nil
}
```

Reply:
Added a test for the case

Reply:
(we never accept paths longer than 10 items anywhere)

Reply:
👍